testbed/datasenders/stanza.go

```diff
@@ -110,6 +110,8 @@ func (f *FileLogWriter) GenConfigYAMLStr() string {
 	// We are testing stanza receiver here.
 	return fmt.Sprintf(`
   filelog:
+    retry_on_failure:
```

Comment:
> Does this change the performance of the test? I assume that a benchamark that retries on failure would possibly provide different results compared to another one that just fails.

In case of filelog receiver, the downstream errors are not transmitted back to sender, if the exporter backend is down. Hence, I enabled `retry_on_failure` to make sure we receive all the data if we face non-permanent errrors.

> Could we consolidate the different setups we test along with their results in a place and make this available to our users?

yes, we can do that.

Reply:
> Is this change required for this PR? Wouldn't that be better to have this in a standalone PR since this "affects" already existing tests? This way it can be distinguished easier and if required would be reverted on its own.

I think we should make this as a conditional switching. Something like `WithRetry(...)`. 
I'll open up a separate PR for it.

Reply:
@ChrsMark https://github.com/open-telemetry/opentelemetry-collector-contrib/pull/36781. PTAL